### PR TITLE
Handle inline and lacking filename

### PR DESCRIPTION
**Pull request by mkly**
Handle Content-Disposition: inline and attachment without filename

* Add new email test example and test with Content-Disposition: inline.
* Move attachment_info above for loop so it is always defined
* Check if item is inline as well as attachment as these both lack an = character to split on
* Create filename if filename is not specified and write file.
* Update list_attachments with new filename

### CODE REVIEW

**Comment by mkly** [comment]
Bit of an edge case email client [Claws Mail](https://www.claws-mail.org/). Noticed an issue with "Content-Disposition: inline" and included an example in the test case. Thanks for the very useful library and let me know if there's anything here I should adjust. :tropical_drink:

File: unstructured/partition/email.py [hunk 1]
```diff
@@ -163,10 +163,10 @@ def extract_attachment_info(
             cdisp = part["content-disposition"].split(";")
             cdisp = [clean_extra_whitespace(item) for item in cdisp]
 
+            attachment_info = {}
             for item in cdisp:
-                attachment_info = {}
 
-                if item.lower() == "attachment":
+                if item.lower() == "attachment" or item.lower() == "inline":
```
Comment (cragwolfe) [inline unstructured/partition/email.py:169]:
```suggestion
                if item.lower() in ("attachment", "inline"):
```
tiny nit

**Review by cragwolfe — COMMENTED** [review]
Awesome contribution! Please run `make tidy` for lint issues.

File: unstructured/partition/email.py (hunk 1 above)
Comment (mkly) [reply unstructured/partition/email.py]: I considered that myself. Thanks for that feedback. Fixed.

**Comment by mkly** [comment]
Apologies for missing the linting and changelog. I've gone ahead an updated this PR as well a #1110 Thank you for taking the time to review these and especially for the feedback.

**Comment by cragwolfe** [comment]
For sure! Thanks for providing clean tests along with the fixes. Text book! :)

**Review by cragwolfe — APPROVED** [review]
LGTM!

**Comment by cragwolfe** [comment]
@mkly , looks like this PR just needs some conflicts resolved. thanks again for the contribution!
